voltixApp/VoltixApp/View Models/CoinViewModel.swift

```diff
@@ -1,68 +1,71 @@
-        if tx.coin.chain.chainType == ChainType.UTXO {
-            await utxo.fetchBlockchairData(for: tx.fromAddress, coinName: coinName)
-        } else if tx.coin.chain.name.lowercased() == Chain.Ethereum.name.lowercased() {
-            await eth.getEthInfo(for: tx.fromAddress)
-        } else if tx.coin.chain.name.lowercased() == Chain.THORChain.name.lowercased() {
-            tx.gas = "0.02"
-            do{
-                let thorBalances = try await thor.fetchBalances(tx.fromAddress)
-                if let priceRateUsd = CryptoPriceService.shared.cryptoPrices?.prices[Chain.THORChain.name.lowercased()]?["usd"] {
-                    balanceUSD = thorBalances.runeBalanceInUSD(usdPrice: priceRateUsd) ?? "US$ 0,00"
-                }
-                coinBalance = thorBalances.formattedRuneBalance() ?? "0.0"
-            }catch{
-                print("error fetching thorchain balances:\(error.localizedDescription)")
-            }
-        }
-        
-        DispatchQueue.main.async {
-            self.updateState(eth: eth, tx: tx)
-        }
-        isLoading = false
-    }
-    
-    public func updateState(eth: EthplorerAPIService, tx: SendTransaction) {
-        let coinName = tx.coin.chain.name.lowercased()
-        let key = "\(tx.fromAddress)-\(coinName)"
-        
-        if tx.coin.chain.chainType == ChainType.UTXO {
-            balanceUSD = utxo.blockchairData[key]?.address?.balanceInUSD ?? "US$ 0,00"
-            coinBalance = utxo.blockchairData[key]?.address?.balanceInBTC ?? "0.0"
-        } else if tx.coin.chain.chainType == ChainType.EVM {
-            tx.eth = eth.addressInfo
-            if tx.coin.ticker.uppercased() == "ETH" {
-                coinBalance = eth.addressInfo?.ETH.balanceString ?? "0.0"
-                balanceUSD = eth.addressInfo?.ETH.balanceInUsd ?? "US$ 0,00"
-            } else if let tokenInfo = tx.token {
-                balanceUSD = tokenInfo.balanceInUsd
-                coinBalance = tokenInfo.balanceString
-            }
-        }
-    }
+	@Published var isLoading = false
+	@Published var balanceUSD: String? = nil
+	@Published var coinBalance: String? = nil
+	@Published var ethAddressInfo: EthAddressInfo = EthAddressInfo()
+	
+	private var utxo = BlockchairService.shared
+	private let thor = ThorchainService.shared
+	private let eth = EtherScanService.shared
+	
+		//TODO: Something is wrong with this load data
+		//TODO: We need to check because it is called at least 5x
+	func loadData(tx: SendTransaction) async {
+		print("realoading data...")
+		isLoading = true
+		await CryptoPriceService.shared.fetchCryptoPrices()
+		
+		let coinName = tx.coin.chain.name.lowercased()
+		if tx.coin.chain.chainType == ChainType.UTXO {
+			await utxo.fetchBlockchairData(for: tx.fromAddress, coinName: coinName)
+		} else if tx.coin.chain.name.lowercased() == Chain.Ethereum.name.lowercased() {
+			do {
+				print("The loadData for \(tx.coin.ticker)")
+				try await eth.getEthInfo(tx: tx)
```

Comment:
`eth.getEthInfo` , it is trying to get gasprice , nonce as well , which is not necessary , because when I open my wallet , all I want to see is the balance , and price.  

We only need gasprice and nonce if I am trying to send a tx

Reply:
As you suggested, we will need to talk to @Rockindash to move it only to the Send page.